sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -141,6 +146,22 @@ def __init__(self, name, init_config, instances):
         )
 
         # Query declarations
+        self._alwayson_queries = QueryExecutor(
```

Comment:
Can we make do with only a single `QueryExecutor` for all "internal" queries? And we construct the list of queries here, conditional on which ones are enabled, before passing them to it?

Reply:
Oh I like that much better. We cannot do that for everything though because some conditions can only be evaluated at runtime, such as the EngineEdition.